Mf2/Parser.php

```diff
@@ -1090,7 +1090,7 @@ public function parseH(\DOMElement $e, $is_backcompat = false, $has_nested_mf =
 		}
 
 		// Check for u-photo
-		if (!array_key_exists('photo', $return) && !$is_backcompat) {
+		if (!array_key_exists('photo', $return) && !in_array('u-', $prefixes) && !$is_backcompat) {
```

Comment:
Does this also require `!$has_nested_mf` as part of the if statement? If so, lets add it as well as a matching test case.